apps/render_cf.py

```diff
@@ -158,6 +164,45 @@ def render_templates(job_types, security_environment, api_name):
         template_file.with_suffix('').write_text(output)
 
 
+def get_compute_environments(job_types: dict, compute_env_file: Path) -> list[dict]:
+    compute_envs = []
+    compute_env_names = []
+    compute_env_imports = set()
+    for _, job_spec in job_types.items():
+        for task in job_spec['tasks']:
+            compute_env = task['compute_environment']
+            if 'name' in compute_env:
+                name = compute_env['name']
+                if name in compute_env_names:
+                    raise NameError(
+                        f'Compute envs must have unique names but the following is defined more than once: {name}.'
+                    )
```

Comment:
My understanding is that the builtin `NameError` exception is typically raised when an undefined variable is referenced, so I don't think it's intended to be raised in other contexts. I think `ValueError` is most appropriate here, thought we could probably replace the entire `if name in compute_env_names` block with `assert name in compute_env_names` and that would be fine too.
```suggestion
                    raise ValueError(
                        f'Compute envs must have unique names but the following is defined more than once: {name}.'
                    )
```